core/src/test/java/io/jenkins/pluginhealth/scoring/probes/JSR305ProbeTest.java

```diff
@@ -0,0 +1,114 @@
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+        ));
+    }
+
+    @Override
+    JSR305Probe getSpy() {
+        return spy(JSR305Probe.class);
+    }
+
+    @Test
+    void shouldReturnPluginsThatUseDeprecatedAnnotations() throws IOException {
+        final Path repo = Files.createTempDirectory("foo");
+        Path directory = Files.createDirectories(repo.resolve("src/main/java"));
+        final Path javaFileWithAllDeprecatedAnnotation = Files.createFile(directory.resolve("test-class-1.java"));
+        final Path javaFileWithNonnullDeprecatedAnnotation = Files.createFile(directory.resolve("test-class-2.java"));
+        final Path javaFileWithCheckForNullDeprecatedAnnotation = Files.createFile(directory.resolve("test-class-3.java"));
+        final Path javaFileWithNoDeprecatedAnnotation = Files.createFile(directory.resolve("test-class-4.java"));
+        final Path txtFileShouldNotBeFound = Files.createFile(directory.resolve("file.txt"));
+        Files.createFile(directory.resolve("test-dummy-class-should-not-be-returned.java"));
+
+        Files.write(javaFileWithAllDeprecatedAnnotation, List.of(
+            "package test;",
+            "",
+            "import javax.annotation.Nonnull;",
+            "import javax.annotation.CheckForNull;"
+        ));
+
+        Files.write(javaFileWithNonnullDeprecatedAnnotation, List.of(
+            "package test;",
+            "",
+            "import javax.annotation.Nonnull;"
+        ));
+
+        Files.write(javaFileWithCheckForNullDeprecatedAnnotation, List.of(
+            "package test;",
+            "",
+            "import javax.annotation.CheckForNull;"
+        ));
+
+        Files.write(javaFileWithNoDeprecatedAnnotation, List.of(
+            "package test;"
+        ));
+
+        Files.write(txtFileShouldNotBeFound, List.of(
+            "This-file-should-not-returned."
+        ));
+
+        when(ctx.getScmRepository()).thenReturn(repo);
+        when(plugin.getName()).thenReturn("foo");
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "message", "status")
+            .isEqualTo(ProbeResult.failure(JSR305Probe.KEY, "Deprecated imports found at foo plugin for test-class-1.java, test-class-2.java, test-class-3.java"));
+        verify(probe).doApply(any(Plugin.class), any(ProbeContext.class));
+
+    }
+
+    @Test
+    void shouldNotReturnPluginsWithUpdatedAnnotations() throws IOException {
+        final Path repo = Files.createTempDirectory("foo");
+        Path directory = Files.createDirectories(repo.resolve("src/main/java"));
+        final Path javaFileWithUpdatedAnnotation = Files.createFile(directory.resolve("test-class-3.java"));
+        Files.createFile(directory.resolve("test-dummy-class-should-not-be-returned.java"));
+
+        Files.write(javaFileWithUpdatedAnnotation, List.of(
+            "package test;"
+        ));
```

Comment:
Actually, the updated annotations are: `edu.umd.cs.findbugs.annotations.CheckForNull` and `edu.umd.cs.findbugs.annotations.NonNull` if you want to actually check for "updated annotations".

Reply:
I agree, this is not really what you are testing. You are "just" validating that the classes have no deprecated imports. 
It would be great to catch updated imports as well, so we could score (later on) plugins with updated imports better than plugin with no imports which would be scored better than plugins with deprecated imports (updated > normal > deprecated).

Reply:
That's kinda tricky though cause we would need to define if it's legit not too use the annotations?

Reply:
true. Let's keep the "if used, they need to be the new ones" approach.

Reply:
The latter: 

> just test that when the new imports are used the probe does not fail?